include/core/syrec/parser/utils/custom_error_messages.hpp

```diff
@@ -95,7 +96,7 @@ namespace syrec_parser {
             case SemanticError::DeclaredVariableBitwidthTooLarge:
                 return "SEM21";
             case SemanticError::InfiniteLoopDetected:
-                return "SEM22";
+                return "SEM23";
```

Comment:
Seems like an unintentional change to me
```suggestion
                return "SEM22";
```

Reply:
The numbering of the semantic errors has been fixed.